### DISCUSSION TITLE
Default vscode extensions in Docker image?

### DISCUSSION THREAD

**Discussion by ZachNew** [gitpod-io/openvscode-server#122] [Q&A]
Is there an easy way to specify default extensions to install when creating a docker image? I'm building a standardized dev env for a team and they will have requests for some basic extensions to come pre-installed in the image. Anyone have an easy way to accomplish this?

**Comment by MohamedElashri** [comment]
I think the easiest way for a dev env is that you use `gitpod/openvscode-server` image as an image base and then install all packages and extensions you need in your new `Dockerfile` and then build your own image.

**Reply by ZachNew** [reply]
This was my thought as well and at this point I have that mostly setup. Now I'm struggling with how to install extensions programmatically. If I try something like
```
RUN /home/openvscode-server-v1.60.2-linux-x64/bin/code --install-extension ms-python.python
```
I get the following error:
```
Missing VSCODE_IPC_HOOK_CLI
The command '/bin/sh -c /home/openvscode-server-v1.60.2-linux-x64/bin/code --install-extension hediet.vscode-drawio' returned a non-zero code: 1
```
And I haven't been able to find documentation on this error yet. Anyone have a idea?

**Reply by MohamedElashri** [reply]
I got the same error and seems that `VSCODE_IPC_HOOK_CLI` is missing from openvscode-server. 

I think there might be a workaround this by forking this repository, adding your extensions to  [extensions](https://github.com/gitpod-io/openvscode-server/tree/main/extensions). Then follow build process for openvscode-server [here](https://github.com/gitpod-io/openvscode-server/blob/main/docs/development.md). 

Then you can replace your released build from your fork with the official [Dockerfile](https://github.com/gitpod-io/openvscode-releases/blob/main/Dockerfile) and build your base image (or directly add other specs you want there also). 
At least this workaround should work with getting your extensions (which shouldn't be the biggest problem as any user can install them at runtime)
